server/routes/api/reviews.js

```diff
@@ -111,4 +111,34 @@ router.get("/requests/:id", authenticate, async (req, res) => {
   }
 });
 
+router.put("/:id", authenticate, async (req, res) => {
+  const requestId = req.params.id;
+  const { messageId, code, comments, authorId } = req.body;
+
+  try {
+    const userId = req.user.id;
+
+    const singleRequest = await Review.find({
+      _id: { $in: [requestId] },
+      requesterId: { $in: [userId] },
+    });
+
+    if (singleRequest[0]) {
+      singleRequest[0].messages.map((message) => {
+        if (message["_id"] == messageId && message.authorId == authorId) {
+          message.code = code;
+          message.comments = comments;
+        }
+      });
+
+      await singleRequest[0].save();
+    } else {
+      return res.sendStatus(400);
+    }
+    return res.sendStatus(200);
```

Comment:
Nitpicky, but would it make more sense to put return res.sendStatus(200) inside the if block? The logic seems to jump around a bit. Alternatively you could say if (!singleRequest[0]) return res.sendStatus(400), and then below that, have the rest of the code (the .map, the .save(), and the return).

Reply:
Thanks, Updated!